### PR TITLE
Fixes and Updates for the 2D Renderer

### PR DESCRIPTION
**Pull request by unity-cchu**
### Purpose of this PR
This PR has some bug fixes as well as quality of life/development improvements for users

---
### Testing status
Passes with full success on Yamato

**Manual Tests**: What did you do?
- [X] Opened test project + Run graphic tests locally
- [X] Built a player
- [X] Checked new UI names with UX convention
- [X] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [X] C# and shader warnings (supress shader cache to see them)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/2d%252Fbug-fixes2-chris/.yamato%252Fupm-ci-universal.yml%2523All_Universal_trunk/787221/job

### CODE REVIEW

File: com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs [hunk 1]
```diff
@@ -45,23 +54,23 @@ struct LightBlendStyleProps
...
-            foreach (var lightBlendStyle in rendererData.lightBlendStyles)
-            {
-                if (lightBlendStyle.enabled)
-                    count++;
-            }
+            //foreach (var lightBlendStyle in rendererData.lightBlendStyles)
+            //{
+            //    if (lightBlendStyle.enabled)
+            //        count++;
+            //}
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs]: Is there a reason to keep this commented?

File: com.unity.render-pipelines.universal/Editor/2D/Renderer2DUpgrader.cs [hunk 2]
```diff
@@ -71,7 +89,13 @@ static void UpgradeSceneTo2DRenderer()
             }
         }
 
-        [MenuItem("Edit/Render Pipeline/Universal Render Pipeline/2D Renderer/Upgrade Project to 2D Renderer (Experimental)")]
+        [MenuItem("Edit/Render Pipeline/Universal Render Pipeline/2D Renderer/Upgrade Scene to 2D Renderer (Experimental)", true)]
+        static bool UpgradeSceneTo2DRendererValidation()
+        {
+            return Light2DEditorUtility.IsUsing2DRenderer();
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Editor/2D/Renderer2DUpgrader.cs]: I'm curious how this function work. What if the asset has both 2D and Forward renderer?

File: com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs (hunk 1 above)
Comment (unity-cchu) [reply com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs]: No. I have removed this code.

File: com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs [hunk 3]
```diff
@@ -123,6 +111,12 @@ public override void OnInspectorGUI()
             serializedObject.Update();
 
             EditorGUI.BeginChangeCheck();
+
+            
+            EditorGUILayout.PropertyField(m_TransparencySortMode, Styles.transparencySortMode);
+            if(m_TransparencySortMode.intValue == (int)TransparencySortMode.CustomAxis)
+                EditorGUILayout.PropertyField(m_TransparencySortAxis, Styles.transparencySortAxis);
```
Comment (rainsing) [inline com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs]: We should increment the indention level for this.

File: com.unity.render-pipelines.universal/Editor/2D/Renderer2DUpgrader.cs (hunk 2 above)
Comment (rainsing) [reply com.unity.render-pipelines.universal/Editor/2D/Renderer2DUpgrader.cs]: It only looks at the default renderer in the pipeline asset.

File: com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs [hunk 4]
```diff
@@ -186,6 +178,13 @@ public override void OnInspectorGUI()
             EditorGUILayout.PropertyField(m_UseDepthStencilBuffer, Styles.useDepthStencilBuffer);
             EditorGUILayout.PropertyField(m_PostProcessData, Styles.postProcessData);
 
+
+            EditorGUI.BeginChangeCheck();
```
[the remark this answers is not in the document]
Comment (unity-cchu) [reply com.unity.render-pipelines.universal/Editor/2D/Renderer2DDataEditor.cs]: Fixed

File: com.unity.render-pipelines.universal/Runtime/2D/Renderer2DData.cs [hunk 5]
```diff
@@ -48,6 +75,8 @@ public class Renderer2DData : ScriptableRendererData
         [SerializeField, Reload("Runtime/Data/PostProcessData.asset")]
         PostProcessData m_PostProcessData = null;
 
+        [SerializeField, Reload("Textures/2D/Black1x1.jpg")]
+        Texture m_BlackTexture = null;
```
[the remark this answers is not in the document]
Comment (unity-cchu) [reply com.unity.render-pipelines.universal/Runtime/2D/Renderer2DData.cs]: Fixed

File: com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs [hunk 6]
```diff
@@ -8,121 +8,189 @@ namespace UnityEngine.Experimental.Rendering.Universal
...
+        public void SetTransparencySortingMode(Camera camera, out TransparencySortMode sortingMode, out Vector3 sortingAxis)
+        {
+            sortingMode = camera.transparencySortMode;
+            sortingAxis = camera.transparencySortAxis;
+
+            if (camera.transparencySortMode == TransparencySortMode.Default)
+            {
+                TransparencySortMode defaultTransparencySortMode = camera.orthographic ? TransparencySortMode.Orthographic : TransparencySortMode.Perspective;
+                camera.transparencySortMode = m_Renderer2DData.transparencySortMode == TransparencySortMode.Default ? defaultTransparencySortMode : m_Renderer2DData.transparencySortMode;
+                camera.transparencySortAxis = m_Renderer2DData.transparencySortAxis;
```
[the remark this answers is not in the document]
Comment (unity-cchu) [reply com.unity.render-pipelines.universal/Runtime/2D/Passes/Render2DLightingPass.cs]: This is fixed

File: com.unity.render-pipelines.universal/Runtime/2D/Light2D.cs [hunk 7]
```diff
@@ -284,6 +285,20 @@ internal static void SetupCulling(ScriptableRenderContext context, Camera camera
...
         }
 
+        internal static bool IsSceneLit(Camera camera)
+        {
+            for (int layer = 0; layer < Light2DManager.lights.Length; layer++)
+            {
+                List<Light2D> lightList = Light2DManager.lights[layer];
+                for (int lightIndex = 0; lightIndex < lightList.Count; lightIndex++)
+                {
+                    if (lightList[lightIndex].IsLightVisible(camera))
```
[the remark this answers is not in the document]
Comment (unity-cchu) [reply com.unity.render-pipelines.universal/Runtime/2D/Light2D.cs]: Fixed

**Comment by phi-lira** [comment]
Adding @tomzig16 because of I don't know about the android test files in this PR.

**Pull request merged**